CMakeLists.txt

```diff
@@ -40,11 +40,13 @@ endif()
 set(CMAKE_CXX_STANDARD 17 CACHE STRING "The C++ standard whose features are requested to build this target.")
 set(CMAKE_C_STANDARD   11 CACHE STRING "The C standard whose features are requested to build this target.")
 
-if(DEFINED GLIBCXX_USE_CXX11_ABI)
-  if(${GLIBCXX_USE_CXX11_ABI} EQUAL 1)
-    set(CXX_STANDARD_REQUIRED ON)
-    string(APPEND CMAKE_CXX_FLAGS " -D_GLIBCXX_USE_CXX11_ABI=1")
-  else()
+include(cmake/CheckAbi.cmake)
+string(APPEND CMAKE_CXX_FLAGS " -D_GLIBCXX_USE_CXX11_ABI=${GLIBCXX_USE_CXX11_ABI}")
+if(${GLIBCXX_USE_CXX11_ABI} EQUAL 1)
+  set(CXX_STANDARD_REQUIRED ON)
+else()
+  if(${CMAKE_CXX_COMPILER} MATCHES "g\\+\\+$" AND NOT ${CMAKE_CXX_COMPILER} MATCHES "clang\\+\\+$")
+    # clang doesn't recognize this flag
```

Comment:
Shouldn't this entire block be guarded by `if(CMAKE_SYSTEM_NAME STREQUAL "Linux")`?

Reply:
Oh, yeah, you are right. I'll correct this.